### DISCUSSION TITLE
feature discussion: share schema model between frontend and backend

### DISCUSSION THREAD

**Discussion by eugsokolov** [fastapi/full-stack-fastapi-template#1337] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
n/a
```

### Description

I have a feature request/discussion topic that I would like the opinion of others (particularly contributors of this repo).

I have noticed repetition between the backend and frontend model definitions:
https://github.com/fastapi/full-stack-fastapi-template/blob/master/frontend/src/client/models.ts
https://github.com/fastapi/full-stack-fastapi-template/blob/master/backend/app/models.py

I suggest to combine these 2 definitions into one, perhaps using something like [protobuf](https://protobuf.dev/) or a json schema (looks like already [done](https://github.com/fastapi/full-stack-fastapi-template/blob/master/frontend/src/client/schemas.ts) for frontend). This would save some mindless copy/paste time waste and reduce errors and typo issues. It would also make the schema definition cleaner and consistent.

Implementation should be fairly straightforward (nothing new) and easily testable.

### Operating System

macOS

### Operating System Details

n/a

### Python Version

n/a

**Comment by saltie2193** [comment, marked as answer]
There shouldn't be two different definitions. 
Everything in `frontend/src/client/` is (/should be) generated based on the defined endpoints of the backend. Therefore the models used in the fronted are already based on the models  defined and used in the backend.

**Comment by eugsokolov** [comment]
ah got it - thanks. i must have missed that in the documentation. assuming it is [this](https://github.com/fastapi/full-stack-fastapi-template/tree/master/frontend#generate-client)?

**Reply by saltie2193** [reply]
The section 'Generate client' in `frontend/README.md` provides the necessary steps to generate the frontend client based on the backend.
